token/typed.go

```diff
@@ -0,0 +1,59 @@
+package token
+
+import (
+	"fmt"
+	"sort"
+)
+
+// ArgumentsParser defines a parser for the arguments of a typed token.
+// Parsing stops unrecoverably at the first error. The return value of Err must be checked before using the values returned by precedings calls to GetInt.
+type ArgumentsParser interface {
+	// GetInt tries to parse the argument name and returns its integer value or defaultValue it the argument is not found.
+	// The return value is valid only if Err returns nil.
+	GetInt(name string, defaultValue int) int
+	// Err returns the first error encountered by the ArgumentsParser.
+	Err() error
+}
+
+// CreateTyped defines a function to create a typed token
+type CreateTyped func(argParser ArgumentsParser) (Token, error)
+
+// typedLookup is a mapping from typed token names to their creation functions
+var typedLookup = make(map[string]CreateTyped)
```

Comment:
Let's be more specific here since this is a global variable, please name this "typedTokenLookup"